File: dotnet/Vaas/src/Vaas/Messages/AuthenticationResponse.cs
using System.Diagnostics.CodeAnalysis;
using System.Text.Json.Serialization;

namespace Vaas.Messages
{
    public class AuthenticationResponse
    {
        [JsonPropertyName("kind")] 
        public string Kind = "AuthResponse";
        
        [JsonPropertyName("success")] 
        public bool Success { get; init; }

        [JsonPropertyName("session_id")] 
        public string? SessionId { get; init; }

        [JsonPropertyName("text")] 
        public string? Text { get; init; }

        [MemberNotNullWhen(true, nameof(SessionId), nameof(Text))]
        public bool IsValid => !string.IsNullOrWhiteSpace(SessionId)
                               && !string.IsNullOrWhiteSpace(Text);
    }
}

File: typescript/src/messages/authentication_response.ts
import { JsonProperty, JsonObject } from "typescript-json-serializer";
import { Message, Kind } from "./message";

@JsonObject()
export class AuthenticationResponse extends Message {
  public constructor(session_id: string, success: boolean, text: string) {
    super(Kind.AuthResponse);
    this.session_id = session_id;
    this.success = success;
    this.text = text;
  }

  @JsonProperty() public session_id: string;
  @JsonProperty() public success: boolean;
  @JsonProperty() public text: string;
}
